### PR TITLE
Reorder steps

### PR DESCRIPTION
**Pull request by javidhsueh**
Few changes:
1. Now we only have the following demo code:
    - 0-starting-code:  empty scaffold
    - 1-basic-graph: 70% done of the app.js (the last 30% will be filled by the students), and bare graph-render.js.
    - 2-graph-layout: complete code of the graph application with layout engine
    - 3-final-version: complete code of interactive graph 

2. Update content, add architecture images.

### CODE REVIEW

File: src/demos/graph/1-basic-graph/src/app.js [hunk 1]
```diff
@@ -1,17 +1,68 @@
...
+      graph: new Graph()
+    };
+  }
+
+  componentDidMount() {
+    this.processData();
+  }
+
+  processData() {
+    if (sampleGraph) {
```
Comment (jckr) [inline src/demos/graph/1-basic-graph/src/app.js]: i'd still remove it ;) will do that too in the building-a-geospatial app

File: src/demos/graph/1-basic-graph/src/app.js [hunk 2]
```diff
@@ -1,17 +1,68 @@
...
+    if (this.state.graph.isEmpty()) {
+      return null;
+    }
+
+    // add GraphRender component here
     return (
       <div>
         <div className='intro'>
-          Empty App, Edit Me!
+          Graph is loaded in the state.
```
Comment (jckr) [inline src/demos/graph/1-basic-graph/src/app.js]: maybe {JSON.stringify(this.state)} ?

**Comment by jckr** [comment]
looking great. FYI, I wrote a more detailed setup guide for folks who don't have a development environment setup, you may want to have links to it for people who may have issues with setup. (that section could also use other pairs of eyes).

File: src/demos/graph/1-basic-graph/src/app.js (hunk 1 above)
Comment (javidhsueh) [reply src/demos/graph/1-basic-graph/src/app.js]: will do

File: src/demos/graph/1-basic-graph/src/app.js (hunk 2 above)
Comment (javidhsueh) [reply src/demos/graph/1-basic-graph/src/app.js]: I tried and it shows a big mess (lots of nodes and edges)

**Pull request merged**